File: backend/api/src/main/scala/com/github/ai/split/api/TimestampDto.scala
package com.github.ai.split.api

import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

case class TimestampDto(
  timestampSeconds: Long,
  formatted: String
)

object TimestampDto {
  implicit val encoder: JsonEncoder[TimestampDto] = DeriveJsonEncoder.gen[TimestampDto]
  implicit val decoder: JsonDecoder[TimestampDto] = DeriveJsonDecoder.gen[TimestampDto]
}

File: backend/api/src/main/java/com/github/ai/split/api/TimestampDto.java
package com.github.ai.split.api;

public class TimestampDto {

    public long timestampSeconds = 0L;
    public String formatted = "";

    public TimestampDto() {}

    public TimestampDto(long timestampSeconds, String formatted) {
        this.timestampSeconds = timestampSeconds;
        this.formatted = formatted;
    }
}